### DISCUSSION TITLE
反射代码生成

### DISCUSSION THREAD

**Discussion by car520120** [xmake-io/xmake#2800] [Q&A]
### 你在什么场景下需要该功能？

Project A 编译前需要先编译 Project B 项目
然后再利用 B 生成 A需要的反射文件(CPP)
最后再编译 Project A 

### 描述可能的解决方案

xmake 目前不知道怎么处理，有一点类似于qt moc 一样的功能，只不过这工具是自己开发的

**Comment by waruqi** [comment]
参考 luajit 包，就是先 build host target，然后再用它生成代码编译。
https://github.com/xmake-io/xmake-repo/blob/bcd6ba140f11307da5840feb7f81ec54eafc574f/packages/l/luajit/port/xmake.lua#L62

另外 `set_policy("build.across_targets_in_parallel", false)` 得配置上，避免受到跨 target 并行编译影响。
